### PR TITLE
[AMORO-2842] Support list all paimon branch

### LINKED ISSUE #2842: [Feature]: Support list all paimon branch
### Description

Currently ams can display iceberg tag and branch, paimon tag. 
Paimon branch information should be displayed

### Use case/motivation

Paimon branch information should be displayed

### Describe the solution

1. Upgrade Paimon to version 0.8, which improves the branch api.

2. Implement ```PaimonTableDescriptor#getTableBranches```

### Subtasks

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by huyuanfeng2018**
## Why are the changes needed?

Close #2842.

## Brief change log

- Support list all paimon branch

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by huyuanfeng2018** [comment]
Paimon's branch is not yet perfect, and we will continue to push this feature after it is perfected.

**Comment by huyuanfeng2018** [comment]
image
Works fine in my local area

File: amoro-format-hudi/src/main/java/org/apache/amoro/formats/hudi/HudiTableDescriptor.java [hunk 1]
```diff
@@ -246,7 +248,7 @@ public List<AmoroSnapshotsOfTable> getSnapshots(
 
   @Override
   public List<PartitionFileBaseInfo> getSnapshotDetail(
-      AmoroTable<?> amoroTable, String snapshotId) {
+      AmoroTable<?> amoroTable, String snapshotId, @Nullable String ref) {
```
Comment (czy006) [inline amoro-format-hudi/src/main/java/org/apache/amoro/formats/hudi/HudiTableDescriptor.java:251]: What about MixedAndIcebergTableDescriptor to use ref?

**Review by czy006 — APPROVED** [review]
LGTM

**Pull request merged**